ports/geckolib/properties.mako.rs

```diff
@@ -347,12 +348,14 @@ ${impl_split_style_coord(ident,
                                         &mut self.gecko.${y_union_ffi_name});
     }
     fn copy_${ident}_from(&mut self, other: &Self) {
-        debug_assert_unit_is_safe_to_copy(self.gecko.${x_unit_ffi_name});
-        debug_assert_unit_is_safe_to_copy(self.gecko.${y_unit_ffi_name});
+        unsafe { self.gecko.${x_union_ffi_name}.reset(&mut self.gecko.${x_unit_ffi_name}) };
+        unsafe { self.gecko.${y_union_ffi_name}.reset(&mut self.gecko.${y_unit_ffi_name}) };
         self.gecko.${x_unit_ffi_name} = other.gecko.${x_unit_ffi_name};
         self.gecko.${x_union_ffi_name} = other.gecko.${x_union_ffi_name};
         self.gecko.${y_unit_ffi_name} = other.gecko.${y_unit_ffi_name};
         self.gecko.${y_union_ffi_name} = other.gecko.${y_union_ffi_name};
+        unsafe { self.gecko.${x_union_ffi_name}.addref_opt(&self.gecko.${x_unit_ffi_name}) };
+        unsafe { self.gecko.${y_union_ffi_name}.addref_opt(&self.gecko.${y_unit_ffi_name}) };
```

Comment:
Similarly here I think it would encapsulate the behaviour better to have a `copy_from` function on `nsStyleUnion` that deals with addrefing if needed.